### DISCUSSION TITLE
[TODO] Add FSRS to https://github.com/TheAlgorithms/Python

### DISCUSSION THREAD

**Discussion by Expertium** [open-spaced-repetition/free-spaced-repetition-scheduler#31] [Ideas]
https://github.com/TheAlgorithms/Python is a repo that has [all kinds of algorithms](https://github.com/TheAlgorithms/Python/blob/master/DIRECTORY.md), from sorts and bit manipulation to Xgboost Classifier and LSTM. Currently, it has 189k stars and is watched by almost 6k people, so it's safe to say that it's famous (among nerds, that is).
![image](https://github.com/user-attachments/assets/45eb2ce9-4806-4637-aa33-e9bc175dac01)
I think adding FSRS there would be nice to boost its popularity.

[Btw, turns out this is 15th most starred repo in the world](https://github.com/EvanLi/Github-Ranking/blob/master/Top100/Top-100-stars.md).
![image](https://github.com/user-attachments/assets/a3222cfb-c6b9-450b-8efa-4e16d29f537d)

**Comment by NamelessGO** [comment]
Others languages can also be made, using https://github.com/open-spaced-repetition/awesome-fsrs as reference:

## [Python](https://github.com/TheAlgorithms/Python) (189k :star:)
- Scheduler (v5): [py-fsrs](https://github.com/open-spaced-repetition/py-fsrs)
- Scheduler (v5): [rs-fsrs-python](https://github.com/open-spaced-repetition/rs-fsrs-python)
- Optimizer: [fsrs-optimizer](https://github.com/open-spaced-repetition/fsrs-optimizer)
- [Deprecated] Optimizer: [fsrs-optimizer-tiny](https://github.com/open-spaced-repetition/fsrs-optimizer-tiny)

## [Rust](https://github.com/TheAlgorithms/Rust) (22k :star:)
- Scheduler (v5): [rs-fsrs](https://github.com/open-spaced-repetition/rs-fsrs)
- Scheduler (v5) + Optimizer: [fsrs-rs](https://github.com/open-spaced-repetition/fsrs-rs)
- Run in browsers: [fsrs-browser](https://github.com/open-spaced-repetition/fsrs-browser)

## [TypeScript](https://github.com/TheAlgorithms/TypeScript) (1.4k :star:)
Scheduler (v5): [ts-fsrs](https://github.com/open-spaced-repetition/ts-fsrs)

## [Go](https://github.com/TheAlgorithms/Go) (16k :star:)
Scheduler (v5): [go-fsrs](https://github.com/open-spaced-repetition/go-fsrs)

## [Java](https://github.com/TheAlgorithms/Java) (19k :star:)
Scheduler (v5): [rs-fsrs-java](https://github.com/open-spaced-repetition/rs-fsrs-java)

## [C](https://github.com/TheAlgorithms/C) (19k :star:) & [C++](https://github.com/TheAlgorithms/C-Plus-Plus) (30k :star:)
Scheduler (v5): [rs-fsrs-c](https://github.com/open-spaced-repetition/rs-fsrs-c)

## Node.js (🚫 No Repo available)
Scheduler (v5): [rs-fsrs-nodejs](https://github.com/open-spaced-repetition/rs-fsrs-nodejs)

## [Dart](https://github.com/TheAlgorithms/Dart) (1.9k :star:)
Scheduler (v4.5): [dart-fsrs](https://github.com/open-spaced-repetition/dart-fsrs)

## [Swift](https://github.com/TheAlgorithms/Swift) (470 :star:)
Scheduler (v4.5): [swift-fsrs](https://github.com/open-spaced-repetition/swift-fsrs)

## [Clojure](https://github.com/TheAlgorithms/Clojure) (87 :star:)
Scheduler (v4): [cljc-fsrs](https://github.com/open-spaced-repetition/cljc-fsrs)

## [Ruby](https://github.com/TheAlgorithms/Ruby) (1.2k :star:)
Scheduler (v4): [rb-fsrs](https://github.com/open-spaced-repetition/rb-fsrs)

## [Kotlin](https://github.com/TheAlgorithms/Kotlin) (1.4k :star:)
Scheduler (v4): [android-fsrs](https://github.com/open-spaced-repetition/android-fsrs)

## [Elixir](https://github.com/TheAlgorithms/Elixir) (233 :star:)
Scheduler (v4): [ex_fsrs](https://github.com/open-spaced-repetition/ex_fsrs)

**Comment by L-M-Sherlock** [comment]
FSRS is not basic as other algorithms added in that repo. I haven't find out an appropriate category to put FSRS in.
